python/lsst/source/injection/combine.py

```diff
@@ -0,0 +1,323 @@
+        catalog_dict, tract = self._get_catalogs(inputs, input_refs)
+        outputs = self.run(catalog_dict, inputs["skyMap"], tract)
+        butler_quantum_context.put(outputs, output_refs)
+
+    def _get_catalogs(
+        self,
+        inputs: dict,
+        input_refs: InputQuantizedConnection,
+    ) -> tuple[dict, int]:
+        """Organize all catalogs in a dictionary with photometry band keys."""
+        catalog_dict = {}
+        tracts = set()
+        for ref, catalog in zip(input_refs.catalogs, inputs["catalogs"]):
+            band = ref.dataId.band.name
+            if band not in catalog_dict:
+                catalog_dict[band] = []
+            # load the patch number to check for patch overlap duplicates later
+            catalog["patch"] = ref.dataId.patch.id
+            catalog_dict[band].append(catalog)
+            tracts.add(ref.dataId.tract.id)
+        # vstack all the catalogs for each band
+        for band, catalog_list in catalog_dict.items():
+            catalog_dict[band] = vstack(catalog_list)
+        if len(tracts) != 1:
+            raise RuntimeError(f"len({tracts=}) != 1")
+        return (catalog_dict, list(tracts)[0])
+
+    def _remove_duplicates(
+        self,
+        catalog: Table,
+        tractInfo,
+    ) -> Table:
+        """Remove tract and patch overlap duplicates."""
+        # TODO: add isPatchInner column
+        duplicates = []
+        for ind, row in enumerate(catalog):
+            tractInner = tractInfo.getInnerSkyPolygon()
+            # check if the source is within inner tract bounds
+            if tractInner.contains(np.deg2rad(row[self.config.col_ra]), np.deg2rad(row[self.config.col_dec])):
+                spherePoint = SpherePoint(
+                    row[self.config.col_ra] * degrees, row[self.config.col_dec] * degrees
+                )
+                # get the patch number from the source's ra,dec
+                patchInfo = tractInfo.findPatch(spherePoint)
+                # check against the patch column
+                if row["patch"] != patchInfo.sequential_index:
+                    duplicates.append(ind)
+            # if the source is not within inner tract bounds, remove it
+            else:
+                duplicates.append(ind)
+        catalog.remove_rows(duplicates)
+        return catalog
+
+    # TODO: implement support for combining multi-component objects on DM-42433
+    def _combine_multi_component_objects(
+        self,
+        catalog: Table,
+        tractInfo,
+        match_radius: float,
+    ) -> Table:
+        # TODO: when creating an injection catalog, make a groupId column to
+        # accurately identify multi-component objects. Use that here instead
+        # matching.
+        with Matcher(catalog[self.config.col_ra], catalog[self.config.col_dec]) as m:
+            idx, selfMatchIndices, _, dists = m.query_self(
+                match_radius,
+                min_match=2,
```

Comment:
What does min_match do?

Reply:
min_match determines the minimum number of matches a source needs to show up in the matched catalog. Matching to itself counts, so min_match=2 requires it to match to some other source. The multi-component stuff is still a draft, so maybe I should remove this whole function and set it up on a new ticket.

Reply:
I've removed any appearances of combine_multi_component_objects and moved them to [DM-42433](https://jira.lsstcorp.org/browse/DM-42433)